### DISCUSSION TITLE
Issue with redirects not being followed

### DISCUSSION THREAD

**Discussion by grahamhar** [seleniumbase/SeleniumBase#1094] [Q&A] [answered]
Hi,

First thanks for your work on this library.

I have an issue which could just be of my own misunderstanding, or the setup of the Docker image I am using for testing with. 

Docker images comes prebuilt with:

> Google Chrome 94.0.4606.81
ChromeDriver 94.0.4606.61 (418b78f5838ed0b1c69bb4e51ea0252171854915-refs/branch-heads/4606@{#1204})

When the test below is run locally on MacOS in headless mode I get a passing test, when I execute this on the AWS Codebuild docker image it looks like the call to open URL is not following the redirect, I think this is proved by the fact the URI of the current browser is still the same as the one it was opened with but it should have redirected to the oauth login URI. Any help or pointers on how to address this would be appreciated.

```python
def test_login(sb, setup_config):
    print(f'heading to {setup_config["landing_uri"]}')
    sb.open(setup_config['landing_uri'])
    url_before_login = sb.get_current_url()
    print(f'XXXXXX{sb.get_page_source()}XXXXX')
    print(landing_uri: {setup_config["landing_uri"]}')
    print(f'before login: {url_before_login}') # this should pring the redirected URI but it is still the same as landing_uri
    sign_in_form_username = sb.find_visible_elements('input#signInFormUsername')
    sign_in_form_username[0].send_keys(setup_config['username'])
    sign_in_form_password = sb.find_visible_elements('input#signInFormPassword')
    sign_in_form_password[0].send_keys(setup_config['password'])
    sign_in_form_button = sb.find_visible_elements('input#signInFormPassword')
    sign_in_form_button[0].submit()
    url_after_login = sb.get_current_url()
    assert url_before_login.startswith(f'https://{setup_config["prefix"]}-login.auth')
    assert url_after_login.startswith(f'https://{setup_config["prefix"]}-app')


```

**Comment by mdmintz** [comment, marked as answer]
Hi @grahamhar Is there any type of exception message, etc, when you're running in Docker? Does the screenshot taken from the failing test show you what happened?
There is an existing Dockerfile with SeleniumBase if that helps: https://github.com/seleniumbase/SeleniumBase/blob/master/Dockerfile
(maybe the Dockerfile you used before did something unusual)

**Reply by grahamhar** [reply]
I’ve done a bit more digging and I think the issue is related to proxy use. I’ve tried setting the standard environment variables no_proxy  for the domain that should not go via proxy but it seems to have no effect, I did see some options on chromedriver but setting the option as a pytest option results in an error of unknown option.

I’ll do some more digging tomorrow and report back.

**Reply by mdmintz** [reply]
@grahamhar  If you're wondering about SeleniumBase proxy use:

#### Run tests through a proxy server:
``pytest --proxy=IP_ADDRESS:PORT``

#### Run tests through a proxy server with authentication:
``pytest --proxy=USERNAME:PASSWORD@IP_ADDRESS:PORT``

**Reply by grahamhar** [reply]
@mdmintz Thanks, I had seen those and was attempting to use them. However, I have a bit of an edge case where the first request shouldn't use a proxy, but the redirect that comes back should use the proxy which is why I am trying to get access to the other option that allows bypass for given domain.

I'm going to work some more in this today and if needed I will attempt to put a PR together

**Reply by grahamhar** [reply]
I've thrown together a PR which seems to fix my immediate needs but probably needs some work before it would be ready to merge, https://github.com/seleniumbase/SeleniumBase/pull/1109

**Reply by mdmintz** [reply]
@grahamhar I added comments to your PR
